src/libyang.h.in

```diff
@@ -1114,8 +1114,9 @@ struct ly_ctx *ly_ctx_new_ylmem(const char *search_dir, const char *data, LYD_FO
  *
  * @param[in] ctx Context to be modified.
  * @param[in] search_dir New search path to add to the current paths previously set in ctx.
+ * @return LY_SUCCESS, LY_EMEM, LY_INVAL.
```

Comment:
general approach in libyang is to return `EXIT_SUCCESS`/`EXIT_FAILURE` and set `ly_errno` to hold the error code. Note, that `ly_errno` is set automatically by logging an error (which is already done in `ly_ctx_set_searchdir()`). So, `ly_ctx_set_searchdir()` should be changed only to set `ly_errno` in case of `!ctx` and possibly to return `EXIT_FAILURE/EXIT_SUCCESS` instead of void (it is possible to detect error via `ly_errno`, but your approach with return code seems to be more clear).